lib/improver/utilities/ancillary_creation.py

```diff
@@ -28,12 +28,192 @@
+        """
+        This scales a set of alphas from input cubes to range between the
+        minimum and maximum alpha values.
+
+        Args:
+            raw_alphas : iris.cube.Cubelist
+                A list of alpha cubes that we need to take the cube_max and
+                cube_min from.
+            min_output : float
+                The minimum value we want our alpha to be
+            max_output : float
+                The maximum value we want our alpha to be
+
+        Returns:
+            scaled_cubes : iris.cube.CubeList
+                A list of alpha cubes scaled to within the range specified.
+        """
+        cube_min = min([cube.data.min() for cube in cubes])
+        cube_max = max([cube.data.max() for cube in cubes])
+
+        scaled_cubes = iris.cube.CubeList()
+        for cube in cubes:
+            scaled_data = (cube.data - cube_min) / (cube_max - cube_min)
+            scaled_data = scaled_data * (max_output - min_output) + min_output
+            scaled_cube = cube.copy(data=scaled_data)
+            scaled_cube.units = '1'
+            scaled_cubes.append(scaled_cube)
+        return scaled_cubes
+
+    def unnormalised_alphas(self, gradient_cube):
+        """
+        This generates initial alpha values from gradients using a generalised
+        power law, whose parameters are set at initialisation.  Current
+        defaults give an output alphas_cube equal to the input gradient_cube.
+
+        Args:
+            gradient_cube : iris.cube.Cube
+                A cube of the normalised gradient
+
+        Returns:
+            alphas_cube : iris.cube.Cube
+                The cube of initial unscaled alphas
+        """
+        alphas_cube = gradient_cube.copy(data=self.coefficient *
+                                         gradient_cube.data**self.power)
+        return alphas_cube
+
+    @staticmethod
+    def update_alphas_metadata(alphas_cube):
+        """
+        Update metadata in alphas cube.  Remove any time coordinates and
+        rename.
+
+        Args:
+        alphas_cube : iris.cube.Cube
+            A cube of alphas with "gradient" metadata
+
+        Returns:
+        alphas_cube : iris.cube.Cube
+            A cube of alphas with adjusted metadata
+        """
+        alphas_cube.rename('alphas')
+        for coord in alphas_cube.coords(dim_coords=False):
+            if 'time' in coord.name() or 'period' in coord.name():
+                alphas_cube.remove_coord(coord)
+        return alphas_cube
+
+    def gradient_to_alpha(self, gradient_x, gradient_y):
+        """
+        Generate alpha smoothing parameters from orography gradients in the
+        x- and y- directions
+
+        Args:
+            gradient_x : iris.cube.Cube
+                A cube of the normalised gradient in the x direction
+            gradient_y : iris.cube.Cube
+                A cube of the normalised gradient in the y direction
+
+        Returns:
```

Comment:
Returns statements that have multiple returns are a tuple of quantities. See the guidance for examples:
https://github.com/metoppv/improver/wiki/Code-Style-Guide#docstrings